site/utils.js

```diff
@@ -38,100 +38,136 @@ const stores = {
+ */
 function currentDate() {
-    const currentDate = new Date();
-    const year = currentDate.getFullYear();
-    const month = String(currentDate.getMonth() + 1).padStart(2, '0');
-    const day = String(currentDate.getDate()).padStart(2, '0');
-    return `${year}-${month}-${day}`;
+  return new Date().toISOString().slice(0, 10);
 }
 
+/**
+ * @description Gets the query parameter from the URL
+ * @param {string} name Name of the query parameter
+ * @returns {string | null} Value of the query parameter or null if not found
+ */
 function getQueryParameter(name) {
-    const url = window.location.href;
-    const queryString = url.substring(url.indexOf('?') + 1);
-    const parameters = queryString.split('&');
-
-    for (var i = 0; i < parameters.length; i++) {
-        const parameter = parameters[i].split('=');
-        const paramName = decodeURIComponent(parameter[0]);
-        if (paramName == name) return decodeURIComponent(parameter[1]);
-    }
-    return null;
+  const url = new URL(window.location.href);
+  return url.searchParams.get(name);
 }
 
+/**
+ * @description Converts a string to a number
+ * @param {string} value String to convert
+ * @param {number} defaultValue Default value if conversion fails
+ * @returns {number} Converted number or default value
+ */
 function toNumber(value, defaultValue) {
-    try {
-        return Number.parseFloat(value);
-    } catch (e) {
-        return defaultValue;
-    }
+  try {
+    return Number.parseFloat(value);
+  } catch (e) {
+    return defaultValue;
+  }
 }
 
+/**
+ * @description Create dom element from html string and add inner html via string template
+ * @param {string} el Element type
+ * @param {string} html Inner html
+ * @returns {HTMLElement} DOM element
+ */
 function dom(el, html) {
-    let element = document.createElement(el);
-    element.innerHTML = html;
-    return element;
+  const element = document.createElement(el);
+  element.innerHTML = html;
+  return element;
 }
 
 async function loadItems() {
-    const response = await fetch("latest-canonical.json")
-    const items = await response.json();
-
-    for (item of items) {
-        item.search = item.name + " " + item.unit;
-        item.search = item.search.toLowerCase().replace(",", ".");
-
-        item.numPrices = item.priceHistory.length;
-        item.priceOldest = item.priceHistory[item.priceHistory.length - 1].price;
-        item.dateOldest = item.priceHistory[item.priceHistory.length - 1].date;
-        item.date = item.priceHistory[0].date;
-        let highestPriceBefore = -1;
-        for (let i = 1; i < item.priceHistory.length; i++) {
-            const price = item.priceHistory[i];
-            highestPriceBefore = Math.max(highestPriceBefore, price.price);
-        }
-        if (highestPriceBefore == -1) highestPriceBefore = item.price;
-        item.highestBefore = highestPriceBefore;
+  const response = await fetch("latest-canonical.json");
```

Comment:
Please use the same indentation as found in the rest of the file(s), i.e. 4 spaces.

Reply:
Sorry habit, was my editor. Changed back and added an config file https://github.com/badlogic/heissepreise/pull/39/commits/84ee1577cc34037a9796af3c84c44a6ee7dd2947